quill-core/src/main/scala/io/getquill/quotation/Parsing.scala

```diff
@@ -288,6 +288,10 @@ trait Parsing {
   private def ident(x: TermName): Ident = identClean(Ident(x.decodedName.toString))
 
   val optionOperationParser: Parser[OptionOperation] = Parser[OptionOperation] {
+    case q"$o.getOrElse[$t]($body)" if is[Option[Any]](o) =>
+      OptionGetOrElse(astParser(o), astParser(body))
+    case q"$o.flatMap[$t]({($alias) => $body})" if is[Option[Any]](o) =>
+      OptionMap(astParser(o), identParser(alias), astParser(body))
```

Comment:
I'm sorry for the late feedback, but it's important to keep the AST as similar to the original quoted code as possible. For SQL, `map`/`flatMap` are translated the same way, but that might not be the case for future extensions of quill.

Reply:
So back to what it was before @mentegy's comment where I had FlatMap and Flatten in the AST? What about Optional? Should I also add that back to the AST as well? I see how you have a good point with this but aaargh! Frustrating to flip code back and fourth like this when maintainers are not on the same page!

Reply:
yeah, I understand your frustration. I think it's something expected, though. People have different options.

@mentegy could you weight in?

Reply:
@deusaquilus Hello, I was not aware of this convention and feel ashamed that made you change your PR changes several times. 
We can merge it as is and release it on your needs. Later, I will update codebase to meet conventions @fwbrasil noticed

Reply:
@mentegy no reason to be ashamed at all! Thank you for all your hard work and contributions to Quill :)

I'm ok with merging and fixing later if it unblocks @deusaquilus

Reply:
Hi @mentegy, no worries! I'll restore back to the version of the code that had FlatMap and Flatten in the AST so at least we can get that part right. We can open up a new issue to discuss if we want an SomeOptional/NoneOptional in the AST.